packages/blade/src/hooks/useId.ts

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable @typescript-eslint/prefer-nullish-coalescing */
+/* eslint-disable @typescript-eslint/no-shadow */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import * as React from 'react';
+import { useSafeLayoutEffect } from './useSafeLayoutEffect';
+
+/**
+ * Credit: https://github.com/reach/reach-ui/blob/develop/packages/auto-id/src/index.tsx
+ */
+let handoffComplete = false;
```

Comment:
I don't know. I've attached the credit on the comment for the original source. 

I assume handoffComplete here tracks if the server has completed the render. Since in `useEffect` doesn't run on server side only client side we can detect if hydration has been done or not. 

```
  React.useEffect(() => {
    if (!handoffComplete) {
      handoffComplete = true; // client render, handoff complete
    }
  }, []);
```

Reply:
This is Confirmed from the source comment. 
https://github.com/reach/reach-ui/blob/d9253130f9255dc8e27a0edd3ce8e6943a444356/packages/auto-id/src/auto-id.ts#L104-L109